pkg/pillar/zedUpload/azureutil/azure.go

```diff
@@ -155,6 +181,60 @@ func DownloadAzureBlob(accountName, accountKey, containerName, remoteFile, local
 	return nil
 }
 
+// DownloadAzureBlobByChunks will process the blob download by chunks, i.e., chunks will be
+// responded back on as and hwen they recieve
+func DownloadAzureBlobByChunks(accountName, accountKey, containerName, remoteFile, localFile string, httpClient *http.Client, prgNotify ChunkStatsChan) error {
+	stats := ChunkStats{}
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, err := container.Exists()
+	if err != nil {
+		return err
+	}
+	if !containerExists {
+		return fmt.Errorf("Container doesn't exist")
+	}
+	blob := container.GetBlobReference(remoteFile)
+	exists, err := blob.Exists()
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("%s: blob doesn't exist", remoteFile)
+	}
+	getErr := blob.GetProperties(nil)
+	if getErr != nil {
+		return getErr
+	}
+	chunkSize := SingleMB
+	stats.Size = int64(blob.Properties.ContentLength)
+	cw := &ChunkWriter{data: stats, prgNotify: prgNotify}
+	readCloser, err := blob.Get(nil)
+	defer readCloser.Close()
+	if err != nil {
+		return err
+	}
+	var processed int64
+	for processed < stats.Size {
+		var rbuf bytes.Buffer
+		bufferSize := math.Min(float64(stats.Size)-float64(processed), float64(chunkSize))
```

Comment:
Converting to float is dangerous; you can get inexact results. PLease do this using integer arithmetic.